ingenialink/dictionary.py

```diff
@@ -51,7 +183,161 @@ def __init__(self, dictionary_path):
+            current_read_register = dict()
+
+            # Identifier
+            current_read_register[self.AttrRegDict.IDENTIFIER] = register.attrib['id']
+
+        except KeyError as ke:
+            logger.error(f"The register doesn't have an identifier. Error caught: {ke}")
+            return None
+
+        try:
+            # Units
+            current_read_register[self.AttrRegDict.UNITS] = register.attrib['units']
+
+            # Cyclic
+            current_read_register[self.AttrRegDict.CYCLIC] = register.attrib.get('cyclic', 'CONFIG')
+
+            # Data type
+            dtype_aux = register.attrib['dtype']
+
+            if dtype_aux in self.__dtype_xdf_options:
+                current_read_register[self.AttrRegDict.DTYPE] = self.__dtype_xdf_options[dtype_aux]
+            else:
+                raise exc.ILValueError(f'The data type {dtype_aux} does not exist for the register: '
+                                       f'{current_read_register["identifier"]}')
+
+            # Access type
+            access_aux = register.attrib['access']
+
+            if access_aux in self.access_xdf_options:
+                current_read_register[self.AttrRegDict.ACCESS] = self.access_xdf_options[access_aux]
+            else:
+                raise exc.ILAccessError(f'The access type {access_aux} does not exist for the register: '
+                                        f'{current_read_register[self.AttrRegDict.IDENTIFIER]}')
+
+            # Subnode
+            current_read_register[self.AttrRegDict.SUBNODE] = int(register.attrib.get('subnode', 1))
+
+            # Storage
+            current_read_register[self.AttrRegDict.STORAGE] = register.attrib.get('storage')
+
+            # Category Id
+            current_read_register[self.AttrRegDict.CAT_ID] = register.attrib.get('cat_id')
+
+            # Description
+            current_read_register[self.AttrRegDict.DESC] = register.attrib.get("desc", 0)
+
+            # Labels
+            labels_elem = register.findall(DICT_LABELS_LABEL)
+            current_read_register[self.AttrRegDict.LABELS] = {label.attrib['lang']: label.text for label in labels_elem}
+
+            # Range
+            range_elem = register.find(self.DICT_RANGE)
+            current_read_register[self.AttrRegDict.REG_RANGE] = (None, None)
+            if range_elem is not None:
+                range_min = range_elem.attrib['min']
+                range_max = range_elem.attrib['max']
+                current_read_register[self.AttrRegDict.REG_RANGE] = (range_min, range_max)
+
+            # Enumerations
+            enums_elem = register.findall(self.DICT_ENUMERATIONS_ENUMERATION)
+            current_read_register[self.AttrRegDict.ENUMS] = [{enum.attrib['value']: enum.text} for enum in enums_elem]
+
+            return current_read_register
+
+        except KeyError as ke:
+            logger.error(f"The register with the ID {current_read_register[self.AttrRegDict.IDENTIFIER]} had some "
```

Comment:
It should be the same text than CANopen and Ethernet:
`logger.error(f"Register with ID {current_read_register[self.AttrRegDict.IDENTIFIER]} has not attribute {ke}")`